PROTO_tests/tests/dtypes_test.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+from arkouda import dtypes
+
+"""
+DtypesTest encapsulates arkouda dtypes module methods
+"""
+
+SUPPORTED_NP_DTYPES = [
+    bool,
+    int,
+    float,
+    str,
+    np.bool_,
+    np.int64,
+    np.float64,
+    np.uint8,
+    np.uint64,
+    np.str_,
+]
+
+
+class TestDTypes:
+    @pytest.mark.parametrize("dtype", SUPPORTED_NP_DTYPES)
+    def test_check_np_dtype(self, dtype):
+        dtypes.check_np_dtype(np.dtype(dtype))
+
+    def test_check_np_dtype_errors(self):
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype(np.dtype(np.int16))
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("np.str")
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("bigint")
+
+    def test_translate_np_dtype(self):
+        for b in [np.bool_, bool]:
+            assert ("bool", 1) == dtypes.translate_np_dtype(np.dtype(b))
+
+        for s in [np.str_, str]:
+            assert ("str", 0) == dtypes.translate_np_dtype(np.dtype(s))
+
+        assert ("int", 8) == dtypes.translate_np_dtype(np.dtype(np.int64))
+        assert ("uint", 8) == dtypes.translate_np_dtype(np.dtype(np.uint64))
+        assert ("float", 8) == dtypes.translate_np_dtype(np.dtype(np.float64))
+        assert ("uint", 1) == dtypes.translate_np_dtype(np.dtype(np.uint8))
+
+    def test_resolve_scalar_dtype(self):
+        assert "bool" == dtypes.resolve_scalar_dtype(True)
+        assert "int64" == dtypes.resolve_scalar_dtype(1)
+        assert "int64" == dtypes.resolve_scalar_dtype(-1)
+        assert "int64" == dtypes.resolve_scalar_dtype(np.int64(1))
+        assert "float64" == dtypes.resolve_scalar_dtype(float(0.0))
+        assert "float64" == dtypes.resolve_scalar_dtype(float(-1.0))
+        assert "float64" == dtypes.resolve_scalar_dtype(np.nan)
+        assert "float64" == dtypes.resolve_scalar_dtype(np.inf)
+        assert "float64" == dtypes.resolve_scalar_dtype(-np.inf)
+        assert "str" == dtypes.resolve_scalar_dtype("test")
+        assert "<class 'list'>" == dtypes.resolve_scalar_dtype([1])
+        assert "uint64" == dtypes.resolve_scalar_dtype(2**63 + 1)
+        assert "bigint" == dtypes.resolve_scalar_dtype(2**64)
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_pdarrays_datatypes(self, size):
+        assert dtypes.dtype("int64") == ak.array(np.arange(size)).dtype
+        assert dtypes.dtype("uint64") == ak.array(np.arange(size), ak.uint64).dtype
+        assert dtypes.dtype("bool") == ak.ones(size, ak.bool).dtype
+        assert dtypes.dtype("float64") == ak.ones(size).dtype
+        assert dtypes.dtype("str") == ak.array([f"string {i}" for i in range(size)]).dtype
```

Comment:
So generating the strings in this way can get really expensive as the `prob_size` grows. A more scalable way is something like
```python
 ak.random_strings_uniform(1, 16, size=size)
```

Reply:
Updated